mlfinlab/labeling/trend_scanning.py

```diff
@@ -5,8 +5,56 @@
 
 import pandas as pd
 import numpy as np
+from numba import njit
 
-from mlfinlab.structural_breaks.sadf import get_betas
+
+@njit
+def calculate_t_values(subset):
+    """
+    Fuor loop for calculating linear regression every n steps.
+    
+    :param subset: (np.array) subset of indecies for which we want to calculate t values
+    :return: (float) maximum t value and index of maximum t value
+    """
+    
+    
+    max_abs_t_value = -np.inf  # Maximum abs t-value of b_1 coefficient among l values
+    max_t_value_index = None  # Index with maximum t-value
+    min_sample_length = 20
+    
+    for forward_window in np.arange(min_sample_length, subset.shape[0], step):
+
+        y = subset[:forward_window].reshape(-1, 1)  # y{t}:y_{t+l}
+
+        # Array of [1, 0], [1, 1], [1, 2], ... [1, l] # b_0, b_1 coefficients
+        X = np.ones((y.shape[0], 2))
+        X[:, 1] = np.arange(y.shape[0])
+
+        # Get regression coefficients estimates
+        xy = X.transpose() @ y
```

Comment:
Can you please explain what is happening here? I am not familiar with the @ syntax.

Reply:
It is another way to write `np.dot( X.T, y)`. I like more this syntax, but can change np no.dot(). For me, it is easier to read: (X.T @ X) @ (X @Y) than np.dot(no.dot(X.T, X), np.dot(X, Y)).